pytorch_lightning/utilities/cli.py

```diff
@@ -48,6 +48,7 @@ def __init__(self, *args, parse_as_dict: bool = True, **kwargs) -> None:
         self.add_argument(
             '--config', action=ActionConfigFile, help='Path to a configuration file in json or yaml format.'
         )
+        self.callbacks = []
 
     def add_lightning_class_args(
```

Comment:
What about adding in a place to override default arugments here?  It would make sense for code locality.  Then something like
```python
def add_lightning_class_args(
        self,
        lightning_class: Union[Type[Trainer], Type[LightningModule], Type[LightningDataModule]],
        nested_key: str,
        class_defaults: Dict[str, Any] = None,
        subclass_mode: bool = False
    ) -> None:
        """
        Adds arguments from a lightning class to a nested key of the parser
        Args:
            lightning_class: Any subclass of {Trainer, LightningModule, LightningDataModule, Callback}.
            nested_key: Name of the nested namespace to store arguments.
            subclass_mode: Whether allow any subclass of the given class.
            class_defaults: Set to override class defaults
        """
```

Reply:
I have added to the example in the docs an override of a default to illustrate how it could be done right now. Adding a `class_defaults` argument to `add_lightning_class_args` is a good idea. But this only makes sense if it works well with the `subclass_mode` option. Certainly can be done, but requires work and I can't do this today.